openupgrade_scripts/scripts/hr_holidays/15.0.1.5/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,198 @@
+hr_holidays  / hr.leave.accrual.plan    / name (char)                   : NEW required
+hr_holidays  / hr.leave.accrual.plan    / time_off_type_id (many2one)   : NEW relation: hr.leave.type
+hr_holidays  / hr.leave.accrual.plan    / transition_mode (selection)   : NEW required, selection_keys: ['end_of_accrual', 'immediately'], req_default: function, hasdefault
+# NOTHING TO DO: new model
+
+hr_holidays  / hr.leave.allocation      / accrual_limit (integer)       : DEL
+hr_holidays  / hr.leave.allocation      / active (boolean)              : NEW hasdefault
+# NOTHING TO DO
+
+hr_holidays  / hr.leave.allocation      / accrual_plan_id (many2one)    : NEW relation: hr.leave.accrual.plan, isfunction: function, stored
+# NOTHING TO DO: handled by ORM
+
+hr_holidays  / hr.leave.allocation      / first_approver_id (many2one)  : DEL relation: hr.employee
+hr_holidays  / hr.leave.allocation      / second_approver_id (many2one) : DEL relation: hr.employee
+hr_holidays  / hr.leave.allocation      / approver_id (many2one)        : NEW relation: hr.employee
+# DONE: pre-migration: map first_approver_id / second_approver_id to approver_id
+
+hr_holidays  / hr.leave.allocation      / date_from (datetime)          : now required, req_default: function
+hr_holidays  / hr.leave.allocation      / date_from (datetime)          : type is now 'date' ('datetime')
+hr_holidays  / hr.leave.allocation      / date_to (datetime)            : not a function anymore
+hr_holidays  / hr.leave.allocation      / date_to (datetime)            : type is now 'date' ('datetime')
+# DONE: pre-migration: convert datetime to date
+
+hr_holidays  / hr.leave.allocation      / employee_company_id (many2one): NEW relation: res.company, isrelated: related, stored
+# DONE: pre-migration: fast fill employee_company_id's value from employee_id.company_id on hr.leave.allocation model
+
+hr_holidays  / hr.leave.allocation      / employee_ids (many2many)      : NEW relation: hr.employee, isfunction: function, stored
+# DONE: pre-migration: create hr_employee_hr_leave_allocation_rel table and fill data
+
+hr_holidays  / hr.leave.allocation      / interval_number (integer)     : DEL
+hr_holidays  / hr.leave.allocation      / interval_unit (selection)     : DEL selection_keys: ['days', 'months', 'weeks', 'years']
+hr_holidays  / hr.leave.allocation      / lastcall (date)               : NEW hasdefault
+hr_holidays  / hr.leave.allocation      / number_per_interval (float)   : DEL
+# NOTHING TO DO: handled by ORM
+
+hr_holidays  / hr.leave.allocation      / multi_employee (boolean)      : NEW isfunction: function, stored
+# DONE: pre-migration: create column and fill data
+
+hr_holidays  / hr.leave.allocation      / state (selection)             : selection_keys is now '['cancel', 'confirm', 'draft', 'refuse', 'validate']' ('['cancel', 'confirm', 'draft', 'refuse', 'validate', 'validate1']')
+# DONE: map validate1 -> confirm
```

Comment:
This doesn't consist only on mapping the value, but also it should clear up the activities. A call to `activity_update` may serve.